crates/cubecl-matmul/src/components/stage/memory/config.rs

```diff
@@ -11,9 +11,47 @@ pub struct StageMemoryConfig {
     pub tiles_in_stage_col: u32,
     pub stage_line_size: u32,
     pub matrix_layout: MatrixLayout,
+    pub swizzle: SwizzleMode,
     pub num_stages: u32,
 }
 
+/// Swizzling mode of the shared memory. Default `None`.
+/// Matches the base TMA functionality, alternative chunk sizes or more complex patterns don't really
+/// apply to matmul.
+#[derive(Default, Hash, PartialEq, Eq, Clone, Debug, Copy)]
+pub enum SwizzleMode {
+    /// No swizzling
+    #[default]
+    None,
+    /// Swizzle 16B chunks within 32B span
+    /// Swizzle<1,4,3>
+    B32,
+    /// Swizzle 16B chunks within 64B span
+    /// Swizzle<2,4,3>
+    B64,
+    /// Swizzle 16B chunks within 128B span
+    /// Swizzle<3,4,3>
+    B128,
+}
```

Comment:
I would add a definition of swizzling atom, the chunks and the span in the SwizzleMode docs. Quite useful.